### PR TITLE
V14: Member group controller

### PR DESCRIPTION
**Pull request by Zeegaan**
# Notes
- Port of `MemberGroupController` to v14
- Implements `async` versions of all the methods in `IMemberGroupService`
- The `DuplicateName` endpoint has now been moved to the service for validation.

# How to test
Using swagger, you should
- [x] Create a member group
- [x] Fetch the created member group
- [x] Update the member group name
- [x] Create another member group
- [x] Fetch all member groups, assert both are there
- [x] Try updating one of the member groups to have the same name as the other, this should fail.
- [x] Try creating a member group with an already existing name (the name of either member group) this should fail.
- [x] Try creating a member group with an already existing key (the key of either member group) this should fail.

### CODE REVIEW

**Comment by Migaroez** [comment]
❗ Updating a group with the same name (not the same as another) gives duplicate name

**Review by Migaroez — CHANGES_REQUESTED** [review]
❗ Updating a group with the same name (not the same as another) gives duplicate name (should just pass)
❗ Deleting a group that does not exist throws a 500 instead of 404

**Review by Migaroez — APPROVED** [review]
Don't forget to update the openapi spec before merging.

File: src/Umbraco.Cms.Api.Management/Controllers/MemberGroup/UpdateMemberGroupController.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        _mapper = mapper;
+        _memberGroupService = memberGroupService;
+    }
+
+    [HttpPut($"{{{nameof(id)}:guid}}")]
+    [MapToApiVersion("1.0")]
+    [ProducesResponseType(typeof(MemberGroupResponseModel), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> Update(Guid id, UpdateMemberGroupRequestModel model)
```
Comment (Migaroez) [inline src/Umbraco.Cms.Api.Management/Controllers/MemberGroup/UpdateMemberGroupController.cs:29]: Missing 404 response type

Comment (Zeegaan) [reply src/Umbraco.Cms.Api.Management/Controllers/MemberGroup/UpdateMemberGroupController.cs:30]: Good catch 👍

**Pull request merged**